test/test_interpolations.jl

```diff
@@ -1,89 +1,200 @@
 using Ferrite: reference_shape_value, reference_shape_gradient
 
-@testset "interpolations" begin
+"""
+    test_interpolation_properties(ip::Interpolation)
 
-    @testset "Value Type $value_type" for value_type in (Float32, Float64)
-        @testset "Correctness of $interpolation" for interpolation in (
-                Lagrange{RefLine, 1}(),
-                Lagrange{RefLine, 2}(),
-                Lagrange{RefQuadrilateral, 1}(),
-                Lagrange{RefQuadrilateral, 2}(),
-                Lagrange{RefQuadrilateral, 3}(),
-                Lagrange{RefTriangle, 1}(),
-                Lagrange{RefTriangle, 2}(),
-                Lagrange{RefTriangle, 3}(),
-                Lagrange{RefTriangle, 4}(),
-                Lagrange{RefTriangle, 5}(),
-                Lagrange{RefHexahedron, 1}(),
-                Lagrange{RefHexahedron, 2}(),
-                Serendipity{RefQuadrilateral, 2}(),
-                Serendipity{RefHexahedron, 2}(),
-                Lagrange{RefTetrahedron, 1}(),
-                Lagrange{RefTetrahedron, 2}(),
-                Lagrange{RefPrism, 1}(),
-                Lagrange{RefPrism, 2}(),
-                Lagrange{RefPyramid, 1}(),
-                Lagrange{RefPyramid, 2}(),
-                #
-                DiscontinuousLagrange{RefLine, 0}(),
-                DiscontinuousLagrange{RefQuadrilateral, 0}(),
-                DiscontinuousLagrange{RefHexahedron, 0}(),
-                DiscontinuousLagrange{RefTriangle, 0}(),
-                DiscontinuousLagrange{RefTetrahedron, 0}(),
-                DiscontinuousLagrange{RefLine, 1}(),
-                DiscontinuousLagrange{RefQuadrilateral, 1}(),
-                DiscontinuousLagrange{RefHexahedron, 1}(),
-                DiscontinuousLagrange{RefTriangle, 1}(),
-                DiscontinuousLagrange{RefTetrahedron, 1}(),
-                DiscontinuousLagrange{RefPrism, 1}(),
-                DiscontinuousLagrange{RefPyramid, 1}(),
-                #
-                BubbleEnrichedLagrange{RefTriangle, 1}(),
-                #
-                CrouzeixRaviart{RefTriangle, 1}(),
-                CrouzeixRaviart{RefTetrahedron, 1}(),
-                RannacherTurek{RefQuadrilateral, 1}(),
-                RannacherTurek{RefHexahedron, 1}(),
-            )
-            # Test of utility functions
-            ref_dim = Ferrite.getrefdim(interpolation)
-            ref_shape = Ferrite.getrefshape(interpolation)
-            func_order = Ferrite.getorder(interpolation)
-            @test typeof(interpolation) <: Interpolation{ref_shape, func_order}
+This function tests the following implementation details for an
+interpolation. All base interpolations should pass this test, but
```

Comment:
Technically we are not checking for "implementation details" but we check for invariants of the interpolation.

Reply:
What do you mean with invariants in this setting? 

I agree that implementation details is perhaps not the best word, but I couldn't come up with something covering all the test cases either... Since we don't check for some properties here, e.g. the Dirac-delta property, interpolation properties seems a bit off, especially since we also check the ArgumentError checks for `reference_shape_value`...

Reply:
We can also just use properties. Invariants are essentially "correctness properties" of functions which hold true under a given program state.

Reply:
OK, let's go with that then. After all, this is just a docstring for a test function (besides, it was already called `test_interpolation_properties` :))

```suggestion
This function tests the following
interpolation properties. All base interpolations should pass this test, but
```